lms/static/js/views/fields.js

```diff
@@ -495,6 +495,205 @@
+                    add: this.fileUploadHandler,
+                    done: this.successHandler,
+                    fail: this.failureHandler
+                });
+            },
+
+            clickedRemoveButton: function () {
+                var self = this;
+                this.hideUserErrorMessage();
+                this.setCurrentStatus('remove');
+                this.showRemovalInProgressMessage();
+                this.showInProgressMessage('Removing...');
+                 $.ajax({
+                    type: 'POST',
+                    url: this.options.profileImageRemoveUrl,
+                    success: function (data, status, xhr) {
+                        self.successHandler();
+                    },
+                    error: function (xhr, status, error) {
+                       self.failureHandler();
+                    }
+                });
+            },
+
+            uploadButtonTitle: function () {
+                if (this.model.has_profile_image()) {
+                    return _.result(this, 'titleEdit')
+                } else {
+                    return _.result(this, 'titleAdd')
+                }
+            },
+
+            removeButtonTitle: function () {
+                return this.titleRemove;
+            },
+
+            successHandler: function (e, data) {
+                var self = this;
+                // Update model to get the latest urls of profile image.
+                this.model.fetch().done(function () {
+                    self.render();
+                    self.setCurrentStatus('');
+                }).fail(function () {
+                    self.showUserErrorMessage(self.errorMessage);
+                });
+            },
+
+            failureHandler: function (e, data) {
+                this.setCurrentStatus('');
+                 if (_.contains([400, 404], data.jqXHR.status)) {
+                    try {
+                        var errors = JSON.parse(data.jqXHR.responseText);
+                        this.showUserErrorMessage(errors.user_message);
+                    } catch (error) {
+                        this.showUserErrorMessage(this.errorMessage);
+                    }
+                } else {
+                    this.showUserErrorMessage(this.errorMessage);
+                }
+                this.render();
+            },
+
+            fileUploadHandler: function (e, data) {
+                if (this.validateImageSize(data.files[0].size)) {
+                    data.formData = {file: data.files[0]};
+                    this.hideUserErrorMessage();
+                    this.setCurrentStatus('upload');
+                    this.showUploadInProgressMessage();
+                    data.submit();
+                }
+            },
+
+            validateImageSize: function (imageBytes) {
+                if (imageBytes < this.options.imageMinBytes) {
+                    this.showUserErrorMessage('Minimum file size not met.');
```

Comment:
I'm  not going to do a full review yet, since this is a WIP, but please remember to mark the user-facing strings in this file for translation.